### ISSUE TITLE
Call for developers or sponsors for streamline prime!

### ISSUE THREAD

**Issue by colleenkhenry** [streamlinevideo/streamline#13]
Hey streamliners

I'm currently working on architecting a streamline-prime project.

It will include...

- Automated cloud deployment and provisioning. 🤖☁
- Faster encoder build / setup ⚡
- A full cloud based / web based workflow for provisioning remotely controllable encoders, and events / embeddable player codes. ☁
- Smaller / rack-able / silent hardware 🖥
- Full hardware / ISP / software / server redundancy ☁☁☁🖥🖥🖥🌐🌐🌐
- Possible HEVC encoded contribution (maybe delivery, depending on how I feel about the matter. It won't be a technical limitation.) 🤮
- SRT protocol contribution (maybe BBR + TCP + Matroska) 😊
- Server side VP9 transcoding (FPGA accelerated on an F1 instance... roughly 30% better than x264 medium) 😍
- Server side H.264 transcoding with X.264 😁
- Multiple origin shields 🛡
- Akamai CDN delivery with QUIC protocol and HTTP 2.0 for non supported clients 🌩
- Fully frame synced (you could cut cables, kill servers, etc, and there are zero glitches on failover.) 🎥
- Sub three second glass-to-glass latency on modern devices (Chrome, Safari, Firefox, Edge, Android Web) with fallback for iOS web to traditional HLS. 🏃💨

Thats where you come in! Are you interested in contributing on the development? 

I could use people well versed in AWS automation, Docker, HTTP origin configuration, HTTP caching optimization, webapp development, etc. We have hardware, FFmpeg development, and streaming video nerdery represented but could also use more of that. 

I'd also be interested in any corporate sponsors that want to step up as well to see this become a reality. 

Thoughts? Anybody down? Hit me up and I'll add you to our Slack.

**Comment by bhperform** [comment]
Any possibility of DRM? WHich would in itself then require Fairplay, Widevine and PlayReady? Although most DRM licence vendors are closed, Intertrust gives you 200 test tokens when you sign up on their pay-as-you-go ExpressPlay service.

**Comment by colleenkhenry** [comment]
@bhperform  I'm open to it but I think that'd be a phase 2 once the video is flowing reliably and in high quality.

**Comment by bhperform** [comment]
> @bhperform I also will admit to not being an expert in DRM. Maybe we should chat it over before I discount it in V1 :)

Some educational material: https://www.intertrust.com/products/drm-system/developer/key-storage/

**Comment by tchakabam** [comment]
@colleenkhenry Happy to share support in the DRM parts. I have done some end-to-end platform integrations with the Widevine&Playready&Fairplay trio before. 

> Hit me up and I'll add you to our Slack.

Are you on video-dev?

**Comment by colleenkhenry** [comment]
Yes, I am in fact on video-dev. There is a streamline room! Join it and we can discuss there!

**Comment by colleenkhenry** [comment]
#streamline is the channel

**Comment by Murmur** [comment]
Microsoft provides a free Playready Test Server. Use this script to create LAURLs https://github.com/HbbTV-Association/ReferenceApplication/blob/master/tools/test/RegisterDRM_MicrosoftTest.py

This is an example of encryptionKey+KID+LAURL. Anyone can use this key and laurl for packaging and and playready playback testing.
```
"key": "12341234123412341234123412341234",
"kid": "43215678123412341234123412341234",
"playready": "https://test.playready.microsoft.com/service/rightsmanager.asmx?cfg=(kid:header,sl:2000,persist:false,firstexp:60,contentkey:EjQSNBI0EjQSNBI0EjQSNA==)"
```

There we have ClearKeyDRM (Chrome/Firefox+dashjs+laurl ok. IE11 does not support it). Use this script for *clearkeydrm license server* https://github.com/HbbTV-Association/ReferenceApplication/blob/master/tools/test/laurl_ck.php   See this [dashjs clearkey example.](https://m.dtv.fi/dash/index_clearkey2.html?video=21)

**Comment by rosswilson** [comment]
I’m super interested in helping with this. I can offer AWS skills and video streaming experience.

How can I help?

**Comment by colleenkhenry** [comment]
Are you on on the video-dev slack? Join #streamline and we can chat :D!
